compiler/damlc/lib/DA/Cli/Options.hs

```diff
@@ -347,5 +348,6 @@ hlintDisabledOpt = flag' HlintDisabled
     <> help "Disable hlint"
   )
 
-hlintUsageOpt :: Parser (Maybe HlintUsage)
-hlintUsageOpt = optional (hlintEnabledOpt <|> hlintDisabledOpt)
+hlintUsageOpt :: Parser HlintUsage
+hlintUsageOpt = fmap (fromMaybe HlintDisabled) $
+  optional (hlintEnabledOpt <|> hlintDisabledOpt)
```

Comment:
Why do we have `hlintDisabledOpt`? It looks like you cannot use it to overwrite a `hlintEnabledOpt` option that comes earlier in the command line so it seems like a useless flag.

Reply:
I think it's nice style to be able to say `--hlint-disable` explicitly despite the default if nothing is provided or an explicit `--hlint-enable ...` is provided. That is, you can say nothing and get the default or you can be explicit about it. So, while it might seem "useless" its presence is symmetric with `--hlint-enable` and thus avoids a potential principle of least surprise violation for the user.

Reply:
Fair enough but if we add this it should overwrite an `hlint-enable` option that is earlier in the command line imho.

Reply:
I don't agree. The behavior described above makes perfect sense to me. If its contentious then I'd be more inclined to withdraw the `--hlint-disable` flag for the sake of harmony.

Reply:
You prefer the current behavior of having `damlc --hlint-enable DATA_DIR --hlint-disable` behave as if only `--hlint-enable DATA_DIR` was passed? That seems extremely unintuitive to me and contradicts the behavior I’ve gotten used to from other tools, e.g., GHC where you can overwrite previous flags with later flags.

Reply:
No, that's not the current behavior. The current behavior is a parse error. As for overriding previous flags with later flags, that seems to me to be a rabbit hole we'd probably be better off not going down today.

Reply:
Oh I see, thanks. A parse error is reasonable. That said, getting the overwrite behavior isn’t particularly hard:
```
fmap (fromMaybe HLintDisabled . lastMay) $ many (hlintEnabledOpt <|> hlintDisabledOpt)
```
should do the trick but totally fine to do that later or not at all for now.

Reply:
Given that its a parse error does that change your position? If it doesn't, then my preferred option is to remove `--without-hlint`. What do you think?

Reply:
Yep, a parse error is reasonable.

Reply:
Awesome. Thanks @moritzkiefer-da .